src/test/java/com/github/dockerjava/cmd/PullImageCmdIT.java

```diff
@@ -92,25 +94,14 @@ public void testPullImage() throws Exception {
         assertThat(inspectImageResponse, notNullValue());
     }
 
-
-
     @Test
     public void testPullNonExistingImage() throws Exception {
-        // does not throw an exception
-
-        if (getVersion(dockerRule.getClient()).equals(RemoteApiVersion.VERSION_1_24)) {
-            // 1.24 - no exception
-        } else if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_26)) {
-            exception.expect(NotFoundException.class);
-        } else {
-            exception.expect(DockerClientException.class);
-        }
+        //different docker version throws different errors here
+        exception.expect(anyOf(instanceOf(NotFoundException.class), instanceOf(DockerClientException.class)));
```

Comment:
Please restore original logic, original code tests for expected exceptions, not or.

Reply:
There was a bug, in the error handling logic it will always throw an error now. But in my tests docker behaves different also on same versions by just switching to swarm. And again I do not like that, that you must handle different exception types depending on the used docker/swarm version.
If we at least let DockerClientException extend DockerException we can say it will always throw an DockerException independent of the used docker / swarm version.

Reply:
library is transparent to docker api/daemon bugs. We don't do magic, NotFoundException is handled in different way by library users because it means that image not found and user may be wanting to switch to other image, all other errors - 500, 501 etc are not recoverable for application usage. And if swarm and docker behaves differently then it should be library user problem. In fact i don't see a problem, in library you can catch it with `oneEx | secondEx` and tests catches exactly like it is.

Reply:
The case why I do not like this behavior is, that the docker daemon indicates an error asynchronously and we make a docker client exception out of it. 
My expectation is that the DockerClientException is only thrown if a client-side error occurred, not a remote one.